### PR TITLE
Fixes preview not working for unpublished units

### PR DESCRIPTION
**Pull request by rehan99000**
**Description:**
This PR fixes preview of unpublished units raising `404`.

**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-2161

**Reference PR:**
https://github.com/REDHOUSEVE/edx-platform/pull/35/files

### CODE REVIEW

**Review by zubair-arbi — CHANGES_REQUESTED** [review]
@rehan99000 Since we are overriding `PREVIEW_LMS_BASE` in `FEATURES` (via django settings override) for all cloud customers (https://edlyio.atlassian.net/wiki/spaces/BM/pages/713785351/Edly+Cloud+Client+Setup+Guide+on+Multisite+Instance) so can we use that instead of declaring it again in site_config?
Screen Shot 2020-10-27 at 6 47 58 PM


For example:
```python
if settings.FEATURES.get('CUSTOM_COURSES_EDX') == hostname:
    return 'draft-preferred'
```

**Comment by rehan99000** [comment]
> @rehan99000 Since we are overriding `PREVIEW_LMS_BASE` in `FEATURES` (via django settings override) for all cloud customers (https://edlyio.atlassian.net/wiki/spaces/BM/pages/713785351/Edly+Cloud+Client+Setup+Guide+on+Multisite+Instance) so can we use that instead of declaring it again in site_config?

@zubair-arbi we are not overriding `PREVIEW_LMS_BASE` in preview site. I am adding it in the document. we would need to declare it outside of django settings override

**Comment by zubair-arbi** [comment]
> > @rehan99000 Since we are overriding `PREVIEW_LMS_BASE` in `FEATURES` (via django settings override) for all cloud customers (https://edlyio.atlassian.net/wiki/spaces/BM/pages/713785351/Edly+Cloud+Client+Setup+Guide+on+Multisite+Instance) so can we use that instead of declaring it again in site_config?
> 
> @zubair-arbi we are not overriding `PREVIEW_LMS_BASE` in preview site. I am adding it in the document. we would need to declare it outside of django settings override

@rehan99000 I think the settings for Studio, LMS and Preview should be almost same except a few changes. How about we also override `PREVIEW_LMS_BASE` for Preview sites as well?

@ziafazal What do you suggest?

**Comment by rehan99000** [comment]
> > > @rehan99000 Since we are overriding `PREVIEW_LMS_BASE` in `FEATURES` (via django settings override) for all cloud customers (https://edlyio.atlassian.net/wiki/spaces/BM/pages/713785351/Edly+Cloud+Client+Setup+Guide+on+Multisite+Instance) so can we use that instead of declaring it again in site_config?
> > 
> > 
> > @zubair-arbi we are not overriding `PREVIEW_LMS_BASE` in preview site. I am adding it in the document. we would need to declare it outside of django settings override
> 
> @rehan99000 I think the settings for Studio, LMS and Preview should be almost same except a few changes. How about we also override `PREVIEW_LMS_BASE` for Preview sites as well?
> 
> @ziafazal What do you suggest?

That works the same. either we override it or add it simply in the site config out side of django settings override. we just need that value here

**Comment by ziafazal** [comment]
@rehan99000 what if you do it like this ?

```
preview_domain = settings.FEATURES['PREVIEW_LMS_BASE'].split(':')[0]
if preview_domain == hostname:
      return 'draft-preferred'
```

**Comment by rehan99000** [comment]
> @rehan99000 what if you do it like this ?
> 
> ```
> preview_domain = settings.FEATURES['PREVIEW_LMS_BASE'].split(':')[0]
> if preview_domain == hostname:
>       return 'draft-preferred'
> ```

@ziafazal why should we split it on `:`? there will be no `:` in URL

**Comment by zubair-arbi** [comment]
```python
if configuration_helpers.get_value('PREVIEW_LMS_BASE', settings.FEATURES.get('PREVIEW_LMS_BASE')):
    return 'draft-preferred'
```
This way our code will work both for `PREVIEW_LMS_BASE` in site_config (for open source community) and for our case we will be overriding `PREVIEW_LMS_BASE` in `settings.FEATURES`.

**Reference PR:** https://github.com/edly-io/edx_xblock_scorm/pull/4/files#r511814480

**Review by zubair-arbi — APPROVED** [review]
Changes looks good to me. 👍 
@rehan99000 Please test your branch on dev multisits before merging the changes.

@fayyazahmed66 Please make sure to override `PREVIEW_LMS_BASE` in Preview site settings configuration in `FEATURES`.

**Comment by zubair-arbi** [comment]
@rehan99000 Please rebase/squash your commits.

File: common/lib/xmodule/xmodule/modulestore/django.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 import re
 
 from django.conf import settings
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
```
Comment (ziafazal) [inline common/lib/xmodule/xmodule/modulestore/django.py:16]: @rehan99000 we should avoid importing `site_configuration` to include unnecessary dependency.

Comment (zubair-arbi) [reply common/lib/xmodule/xmodule/modulestore/django.py:16]: I agree. @rehan99000 just use `settings.FEATURES.get('PREVIEW_LMS_BASE')`

**Review by zubair-arbi — APPROVED** [review]
LGTM 👍

**Pull request merged**